lib/drives/pages/drive_chat_page.dart

```diff
@@ -16,107 +17,130 @@ class DriveChatPage extends StatefulWidget {
+                }
+              },
+            )
+          : Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              crossAxisAlignment: CrossAxisAlignment.center,
+              children: [
+                Image.asset(
+                  'assets/chat_shrug.png',
+                  scale: 8,
+                ),
+                const SizedBox(height: 16),
+                Text(
+                  S.of(context).pageChatEmptyTitle,
+                  style: Theme.of(context).textTheme.headline6,
+                ),
+                const SizedBox(height: 8),
+                Padding(
+                  padding: const EdgeInsets.symmetric(horizontal: 16),
+                  child: Text(
+                    S.of(context).pageDriveChatEmptyMessage,
+                    textAlign: TextAlign.center,
+                  ),
+                ),
+              ],
+            ),
+    );
+  }
+
+  Widget _buildChatWidget(Ride ride) {
+    ride.messages!.sort((a, b) => b.createdAt!.compareTo(a.createdAt!));
+    Message? lastMessage = ride.messages!.isEmpty ? null : ride.messages!.first;
+    return Card(
+      child: InkWell(
+        child: ListTile(
+          leading: Avatar(ride.rider!),
+          title: Text(ride.rider!.username),
+          subtitle: lastMessage == null ? null : Text(_truncate(lastMessage.content)),
+          trailing: ride.getUnreadMessagesCount() == 0
+              ? null
+              : Container(
+                  decoration: BoxDecoration(color: Theme.of(context).primaryColor, shape: BoxShape.circle),
+                  child: Padding(
+                    padding: const EdgeInsets.all(8.0),
+                    child: Text(
+                      ride.getUnreadMessagesCount().toString(),
+                    ),
+                  ),
+                ),
+          onTap: () {
+            Navigator.of(context)
+                .push(
+                  MaterialPageRoute(
+                    builder: (BuildContext context) => ChatPage(
+                      rideId: ride.id!,
+                      profile: ride.rider!,
+                    ),
+                  ),
+                )
+                .then((value) => setState(() {}));
           },
         ),
       ),
     );
   }
 
-  Widget _riderList(Set<Profile> riders) {
-    Widget ridersColumn = Container();
-    if (riders.isNotEmpty) {
-      ridersColumn = Column(
-        children: List.generate(
-          riders.length,
-          (index) => Padding(
-            padding: const EdgeInsets.symmetric(horizontal: 5.0, vertical: 5.0),
-            child: ProfileWidget(
-              riders.elementAt(index),
-              //TODO go to chat
-              onTap: () {},
-              actionWidget: const Icon(
-                Icons.chat,
-                color: Colors.black,
-                size: 36.0,
-              ),
-            ),
-          ),
-        ),
-      );
+  String _truncate(String text) {
+    int length = 30;
```

Comment:
Are you sure this works with all text sizes? Whatsapp only ever shows exactly two lines, regardless of size. Can we do that somehow?